src/api.c

```diff
@@ -3506,6 +3506,16 @@ int cgroup_get_cgroup(struct cgroup *cgroup)
 			cgroup->tasks_gid = stat_buffer.st_gid;
 
 			free(control_path);
+		} else { /* cgroup v2 */
+			bool enabled;
+
+			ret = cgroupv2_get_subtree_control(path, cg_mount_table[i].name,
+							  &enabled);
+			if (ret)
```

Comment:
@drakenclimber Thank you much for adding the Python bindings/test cases and the patchset looks great.  I have some minor comments.
Should we bail out on the error instead of continuing because the error might be due to `malloc()` or the `cgroup.subtree_control` not readable, given that `CGROUP V2` should always succeed in reading.

Reply:
Yeah, I think that's a good idea.  I think I had a bug or something in an early iteration of this patch, and I put the `continue` in to alleviate it.  It's no longer needed.

Reply:
Actually it was correct.  Controllers can be mounted v2 but not enabled in the subtree file, and thus they'll return `ECGROUPNOTMOUNTED`.

I think this is the right answer (along with some better return codes in `cgroupv2_get_subtree_control()`):
```
+               } else { /* cgroup v2 */
+                       bool enabled;
+
+                       error = cgroupv2_get_subtree_control(path, cg_mount_table[i].name,
+                                                            &enabled);
+                       if (error == ECGROUPNOTMOUNTED)
+                               continue;
+                       if (error)
+                               goto unlock_error;
+
+                       if (!enabled)
+                               continue;
```

Reply:
`cgroupv2_get_subtree_control()` is probably a good candidate for refactoring, though.  It only returns `enabled=1` when the error code is 0.  This is kinda silly and obviously redundant.

Reply:
Agreed, as we spoke, let me take up the error code auditing.